api/buildings/admin.py

```diff
@@ -29,6 +30,50 @@ class BuildingAdmin(admin.ModelAdmin):
         "general_id",
     )
     search_fields = ("address",)
+    actions = (
+        "set_status_to_pending",
+        "set_status_to_accepted",
+        "set_status_to_rejected",
+    )
+
+    def set_status_to_pending(self, request, queryset):
+        updated = queryset.update(status=0)
+        self.message_user(
+            request,
+            ngettext(
+                "%d building was successfully marked as pending.",
+                "%d buildings were successfully marked as pending.",
+                updated,
+            )
+            % updated,
+            messages.SUCCESS,
+        )
```

Comment:
This piece of code just keeps on repeating. Could you move it to a separate method? (e.g. `_message_user(self, request, updated, status)` where `status` is `pending`, `accepted`, or `rejected`).
Also, please use `gettext_lazy` for the status strings

Reply:
I use `ngettext` because is a pluralized message.
Reference: [Pluralization](https://docs.djangoproject.com/en/3.1/topics/i18n/translation/#pluralization)

Reply:
I didn't have anything against `ngettext`. I researched it beforehand and it looked like the right thing for the job, so kudos for that 👌 
I was just suggesting to add a generic method like this. When I first thought this I thought of the status string being passed as a parameter, so it needed translation. Now I realised it can be taken from the existing code. 
```
def _message_user(self, request, queryset, status):
    updated = queryset.update(status=status)
    status_str = BUILDING_STATUS_CHOICES[status]  # not the right code but I hope it gets the point across
    self.message_user(
        request,
        ngettext(
            "%d building was successfully marked as {}.".format(status_str),
            "%d buildings were successfully marked as {}.".format(status_str),
            updated,
        )
        % updated,
        messages.SUCCESS,
    )
```